File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpModData.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;

public record RMcpModData(
        String id,
        String name,
        String version,
        String description,
        List<Dependency> dependencies
) {
    public record Dependency(
            String id,
            String versionRange,
            String type,
            String ordering,
            String side
    ) {
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpModData.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpModData(
    val id: String,
    val name: String,
    val version: String,
    val description: String,
    val dependencies: List<Dependency>
) {
    @JvmRecord
    data class Dependency(
        val id: String,
        val versionRange: String,
        val type: String,
        val ordering: String,
        val side: String?
    )
}
